### PR TITLE
Make homepage companies list responsive on mobiles

### PR DESCRIPTION
**Pull request by rachaelcodes**
## Description of change

The companies list table on the homepage is not currently responsive on mobile, meaning that companies information was not immediately visible. The underlying gov-uk/react Table components did not include breakpoints. 

Using designs from Rogier and an approach suggested by [this article](https://css-tricks.com/responsive-data-tables/), this PR reimplements the Table components with separate styling for mobile and tablet/desktop. The filters above the table are updated so that they stack in the mobile view.

The changes include some additional table cell variants given the design:
* `<AnnotatedCell>` This displays a label for mobile views to replace the hidden row labels. (This could be replaced by passing an optional label prop into the cell component - let me know if you think this would be clearer).
* `<TitleCell>` Adds extra spacing around the 'title' cell of a row in mobile view. (This could be replaced with `:first-child` styling, again, let me know if this seems preferable).
* `<ActionCell>` Makes buttons within a cell full-width in the mobile view, adding extra padding according to the design.

## Test instructions

If you do not have 'My lists' set up on Data Hub,
* go to the 'Companies' view and click on a company title to view its details
* from the company view, click on the 'Add to or remove from lists' button
* you will be prompted to create and name a list; after doing this, return to the company name and add it to one of your created lists

When you have companies added to a custom list on Data Hub, return to the Data Hub page and view the company list table at different viewport widths.

### Before

image

### After

**Updated after discussion with @mforner13**

Screenshot 2020-04-08 at 14 55 59

**previous look**
updated mobile view

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"

- [x] Has the branch been rebased to master?
- [x] Automated tests (Any of the following when applicable: Unit, Functional or Acceptance)
- [x] Manual compatibility testing (Browsers: Chrome, Firefox, IE11, Safari)

### CODE REVIEW

File: src/client/components/CompanyLists/Table.jsx [hunk 1]
```diff
@@ -14,6 +17,75 @@ const StyledCellLink = styled.a({
...
+  [MEDIA_QUERIES.TABLET]: {
+    paddingTop: SPACING.SCALE_2,
+  },
+})
+
+const AnnotatedCell = styled(StyledTableCell)({
+  position: 'relative',
+  marginTop: `${SPACING.SCALE_4}`,
+  '::before': {
+    content: "'Last interaction'",
```
Comment (mforner13) [inline src/client/components/CompanyLists/Table.jsx:65]: Is 'Last interaction' purposefully in a smaller font-size and not in bold anymore? It seems to get a little lost when in mobile view.

File: src/client/components/CompanyLists/Table.jsx [hunk 2]
```diff
@@ -14,6 +17,75 @@ const StyledCellLink = styled.a({
...
+})
+
+const TitleCell = styled(StyledTableCell)({
+  paddingTop: SPACING.SCALE_3,
+  [MEDIA_QUERIES.TABLET]: {
+    paddingTop: SPACING.SCALE_2,
+  },
+})
+
+const AnnotatedCell = styled(StyledTableCell)({
```
Comment (mforner13) [inline src/client/components/CompanyLists/Table.jsx:61]: As this component is only being used in this file, would it be worth calling it something like ColumnLabelCell to make it a bit clearer?

File: src/client/components/CompanyLists/Table.jsx (hunk 1 above)
Comment (rachaelcodes) [reply src/client/components/CompanyLists/Table.jsx:65]: It was suggested in the design, but I could check with Rogier and Billy to see if they'd prefer a change?
invision

Comment (rachaelcodes) [reply src/client/components/CompanyLists/Table.jsx:65]: OK, I've talked to Rogier and he's recommended adding a bit more padding before the label.

**Pull request merged**